news-bundle/src/Resources/contao/models/NewsModel.php

```diff
@@ -244,8 +244,7 @@ public static function findPublishedByPids($arrPids, $blnFeatured=null, $intLimi
 			$arrColumns[] = "$t.featured=''";
 		}
 
-		// Never return unpublished elements in the back end, so they don't end up in the RSS feed
-		if (!static::isPreviewMode($arrOptions) || TL_MODE == 'BE')
+		if (!static::isPreviewMode($arrOptions))
```

Comment:
Instead of duplicating this check multiple times, why not use the scope matcher here?

Reply:
Because checking the scope is wrong here. The `findPublishedByPids()` method should always return the same results in the back end and in the front end. Just like the other `findPublishedByPids()` methods do.

Reply:
I agree in principle, but it still seems wrong to me to duplicate this logic into several places now.

Reply:
we _could_ have a helper method in the model, but then again our models don't contain logic, right?

Reply:
As I said, the model is the **wrong** place to check this! It should be done where the RSS feed is generated.

Also, I don't see a problem with duplicating one single if condition into two separate extensions. Each extension has its own `generateFiles()` method with its own logic, so of course, we could provide a helper method or abstract class, however, that is out of the scope of this PR.

Reply:
This check is already everywhere in the codebase I tink. 

I think streamlining this might be a good idea but should probably not be part of this PR.

Reply:
> I don't see a problem with duplicating one single if condition into two separate extensions.

@leofeyer I do - the condition is fairly complex and its implementation, copying it and its logic over and over again is cumbersome and error prone. Case in point: the implmentation in this PR is wrong: it is missing `Date::floorToMinute()`.

> This check is already everywhere in the codebase I tink.

Mostly in models, but there are some exceptions, yes.

Reply:
> it is missing `Date::floorToMinute()`

This is only used for database queries AFAIK.

Reply:
> This is only used for database queries AFAIK.

Yes - but that's wrong. The PHP check needs to use `Date::floorToMinute()` as well - otherwise you will have different results, i.e. without flooring the element in question would be invisible up to 59 seconds sooner, depending on where you check. We should probably fix this in the other places as well (`SitemapController` and `AbstractContentElementController` for instance).